### ISSUE TITLE
"Unable to find 'mime.types' file in classpath" warning

### ISSUE THREAD

**Issue by shanebishop** [aws/aws-sdk-java#2711]
### Describe the issue

My team has begun to see the following warning level message from our [Kinesis Data Analytics](https://docs.aws.amazon.com/kinesisanalytics/latest/java/what-is.html) application:

```
{
    "locationInformation": "com.amazonaws.services.s3.internal.Mimetypes.getInstance(Mimetypes.java:95)",
    "logger": "com.amazonaws.services.s3.internal.Mimetypes",
    "message": "Unable to find 'mime.types' file in classpath",
    "threadName": "s3a-transfer-shared-pool1-t3",
    "applicationARN": "<redacted>",
    "applicationVersionId": "3",
    "messageSchemaVersion": "1",
    "messageType": "WARN"
}
```

This error message looks like it is coming from [here](https://github.com/aws/aws-sdk-java/blob/cb80ca0b5ba36686ea25b1b13f064af6b2bd2194/aws-java-sdk-s3/src/main/java/com/amazonaws/services/s3/internal/Mimetypes.java#L95) in the SDK.

In my team's `build.gradle`, we pull in the S3 Java SDK like this:

```
implementation "software.amazon.awssdk:s3:2.17.68"
```

What steps (if any) can my team take to identify is this is a false or true positive?

Assuming this warning is not a false positive, how can my team fix this warning?

### Steps to Reproduce

1. Compile an Apache Flink application using Gradle, using `software.amazon.awssdk:s3:2.17.68` as one of the dependencies
2. Create a Kinesis Data Analytics application in AWS
3. Configure the KDA to use the JAR compiled in step 1
4. Run the KDA

### Current behavior

The current behaviour is that a `Unable to find 'mime.types' file in classpath` warning is produced.

### AWS Java SDK version used

2.17.68

### JDK version used

11.0.13

### Operating System and version

Whatever Kinesis Data Analytics uses (presumably some recent version of Linux)

**Comment by debora-ito** [comment]
Hi @shanebishop 

`com.amazonaws.services.s3.internal.Mimetypes.getInstance` is a method from the Java SDK 1.x s3 module, maybe that's the issue? Can you add `com.amazonaws:aws-java-sdk-s3:1.12.159` to the build.gradle and see if the warning goes away?

You mentioned that your team started to see the warning, was there any change that may have triggered it, a code, environment or version change?

**Comment by shanebishop** [comment]
My team is using `software.amazon.awssdk:s3:2.17.68` because we want to use `com.amazonaws.services.s3.S3AsyncClient`. Is there an equivalent async client class in `com.amazonaws:aws-java-sdk-s3:1.12.159`?

I did a quick File Finder search in GitHub and didn't see any clear equivalent class.

**Comment by shanebishop** [comment]
@debora-ito I switched to using `implementation 'com.amazonaws:aws-java-sdk-s3:1.12.159'` in my `build.gradle`, but now I get this error when I compile:

```
/redacted/path/to/S3Downloader.java:5: error: cannot find symbol
import com.amazonaws.services.s3.S3AsyncClient;
                                ^
  symbol:   class S3AsyncClient
  location: package com.amazonaws.services.s3
```

**Comment by shanebishop** [comment]
Adding `com.amazonaws:aws-java-sdk-s3:1.12.159` as a dependency alongside `software.amazon.awssdk:s3:2.17.68` made the warning go away.

Thanks for your help @debora-ito!

**Comment by shanebishop** [comment]
Hi @debora-ito,

I am reopening this issue because I want to better understand this.

It is true that adding `com.amazonaws:aws-java-sdk-s3:1.12.159` as a dependency alongside `software.amazon.awssdk:s3:2.17.68` made the warning go away.

However, what I don't understand is why this change fixes the warning. Both when I compile my JAR with and without `com.amazonaws:aws-java-sdk-s3:1.12.159` I get the same `mime.types` files in the JAR:

```
$ jar -tf $JAR_FILE | grep mime.types
mime.types
org/apache/flink/kinesis/shaded/software/amazon/awssdk/core/util/mime.types
software/amazon/awssdk/core/util/mime.types
```

**Comment by debora-ito** [comment]
The full classpath makes the difference in this case. The class that is missing is being looked for in `com.amazonaws.services.s3`. The `mime.types` files present in your jar have a different classpath. 

Note that I don't know how Kinesis Data Analytics or Apache Flink works, I've suggested you to add the `com.amazonaws:aws-java-sdk-s3` dependency based on the error message - for example, if you were not adding `com.amazonaws:aws-java-sdk-s3` before, then how is Kinesis Data Analytics or Flink using the SDK v1? Where is the error coming from?

```
$ jar -tf compiled-without-s3-sdk.jar | grep mime.types
mime.types
org/apache/flink/kinesis/shaded/software/amazon/awssdk/core/util/mime.types
software/amazon/awssdk/core/util/mime.types
```
⬆️ Other thing I noticed is looks like `software.amazon.awssdk:s3` is already been shaded by Flink, so in theory you don't have to add it as a dependency.
